agent/internal/actor/http.go

```diff
@@ -120,13 +127,43 @@ func (p *blockedUserEventProperties) MarshalJSON() ([]byte, error) {
 func (p *blockedUserEventProperties) GetActionId() string {
 	return p.action.ActionID()
 }
-func (p *blockedUserEventProperties) GetOutput() api.BlockedUserEventProperties_Output {
-	return *api.NewBlockedUserEventProperties_OutputFromFace(p)
+func (p *blockedUserEventProperties) GetOutput() api.BlockedUserEventPropertiesOutput {
+	return *api.NewBlockedUserEventPropertiesOutputFromFace(p)
 }
 func (p *blockedUserEventProperties) GetUser() map[string]string {
 	return p.userID
 }
 
+// redirectedUserEventProperties implements `types.EventProperties` to be marshaled
+// to an SDK event property structure.
+type redirectedUserEventProperties struct {
+	action Action
+	userID map[string]string
+}
+
+// Static assert that redirectedUserEventProperties implements `types.EventProperties`.
+var _ types.EventProperties = &redirectedUserEventProperties{}
+
+func newRedirectedUserEventProperties(action Action, userID map[string]string) *redirectedUserEventProperties {
+	return &redirectedUserEventProperties{
+		action: action,
+		userID: userID,
+	}
+}
+func (p *redirectedUserEventProperties) MarshalJSON() ([]byte, error) {
+	pb := api.NewRedirectedUserEventPropertiesFromFace(p)
```

Comment:
What is `Face`?

Reply:
Every API message can be created from an interface type having the fields as getters. This comes from the time I was generating the API code with protobufs and I kept this idea.